lucene/core/src/java21/org/apache/lucene/store/MemorySegmentIndexInput.java

```diff
@@ -303,6 +304,48 @@ public byte readByte(long pos) throws IOException {
     }
   }
 
+  @Override
+  public void readGroupVInts(long[] dst, int limit) throws IOException {
+    int i;
+    for (i = 0; i <= limit - 4; i += 4) {
+      readGroupVInt(dst, i);
+    }
+    for (; i < limit; ++i) {
+      dst[i] = readVInt();
+    }
+  }
+
+  private void readGroupVInt(long[] dst, int offset) throws IOException {
+    if (curSegment.byteSize() - curPosition < GroupVIntUtil.MAX_LENGTH_PER_GROUP) {
+      GroupVIntUtil.fallbackReadGroupVInt(this, dst, offset);
+      return;
+    }
+
+    try {
+      final int flag = readByte() & 0xFF;
```

Comment:
Since we already did the job of checking that we have enough bytes left, we don't need `readByte()` to do it for us?

```suggestion
      final int flag = curSegment.get(LAYOUT_BYTE, curPosition++) & 0xFF;
```